REGRESSION (iOS 17.4 Beta): No incoming video in Teams VA

youennf opened the pull request:
#### 1392a12ad3f10940ca689b99905c1b8b9d5a089d
<pre>
REGRESSION (iOS 17.4 Beta): No incoming video in Teams VA
<a href="https://rdar.apple.com/122859173">rdar://122859173</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=269281">https://bugs.webkit.org/show_bug.cgi?id=269281</a>

Reviewed by Brent Fulgham.

WebRTC applicartions typically do not care about reordering since it hits the latency.
We are thus setting the window size to 0 when AVC is false, which is the case for libwebrtc decoders.
WebCodecs websites that had issues with reordering are using AVC, so should not be broken by this change.

* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
(-[RTCVideoDecoderH264 decodeData:size:timeStamp:]):

Canonical link: <a href="https://commits.webkit.org/274581@main">https://commits.webkit.org/274581@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/205d983a22d8f76b4a756e30cb96f706c98a895d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39481 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18460 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42015 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/35381 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21339 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15789 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40055 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15554 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/34192 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/13510 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35140 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43293 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35847 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35471 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39273 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14293 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11782 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/37524 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15899 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8831 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/15947 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15556 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
```diff
@@ -161,7 +161,7 @@ - (NSInteger)decodeData:(const uint8_t *)data
       rtc::ScopedCF(webrtc::CreateVideoFormatDescription(data, size));
   if (inputFormat) {
-    _reorderQueue.setReorderSize(webrtc::ComputeH264ReorderSizeFromAnnexB(data, size));
+    _reorderQueue.setReorderSize(_useAVC ? webrtc::ComputeH264ReorderSizeFromAnnexB(data, size) : 0);
```
Please add a FIXME comment to remove this workaround once MS Teams fixes their streaming app.

youennf replied:
Done

brentfulgham commented on Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
```diff
@@ -161,7 +161,8 @@ - (NSInteger)decodeData:(const uint8_t *)data
       rtc::ScopedCF(webrtc::CreateVideoFormatDescription(data, size));
   if (inputFormat) {
-    _reorderQueue.setReorderSize(webrtc::ComputeH264ReorderSizeFromAnnexB(data, size));
+    // FIXME: Reenable window computation size for AnnexB decoders, once feasible.
```
Can you put the relevant Radar or Bugzilla ID in the FIXME?
(i.e., `FIXME(rdar://123456): Reenable window ...`)

youennf replied:
Done

brentfulgham reviewed (approved):
Looks good, but please put the relevant radar in the FIXME so we can find this later.

youennf committed:
REGRESSION (iOS 17.4 Beta): No incoming video in Teams VA
rdar://122859173
https://bugs.webkit.org/show_bug.cgi?id=269281

Reviewed by Brent Fulgham.

WebRTC applicartions typically do not care about reordering since it hits the latency.
We are thus setting the window size to 0 when AVC is false, which is the case for libwebrtc decoders.
WebCodecs websites that had issues with reordering are using AVC, so should not be broken by this change.

* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
(-[RTCVideoDecoderH264 decodeData:size:timeStamp:]):

Canonical link: https://commits.webkit.org/274581@main
